dev-lua/cqueues/cqueues-20200726_p20241204.ebuild

```diff
@@ -0,0 +1,157 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LUA_COMPAT=( lua5-{1..4} luajit )
+
+inherit lua toolchain-funcs
+
+DESCRIPTION="Stackable Continuation Queues"
+HOMEPAGE="https://github.com/wahern/cqueues"
+HOMEPAGE+=" http://25thandclement.com/~william/projects/cqueues.html"
+EGIT_COMMIT="8c0142577d3cb1f24917879997678bef0d084815"
+SRC_URI="https://github.com/wahern/${PN}/archive/${EGIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+
+S="${WORKDIR}/${PN}-${EGIT_COMMIT}"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="examples"
+
+REQUIRED_USE="${LUA_REQUIRED_USE}"
+
+DEPEND="
+	${LUA_DEPS}
+	dev-libs/openssl:0=
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	dev-lua/compat53[${LUA_USEDEP}]
+	virtual/pkgconfig
+"
+
+DOCS=( "doc/." )
+
+lua_src_prepare() {
+	pushd "${BUILD_DIR}" || die
+
+	if [[ ${ELUA} != luajit ]]; then
+		LUA_VERSION="$(ver_cut 1-2 $(lua_get_version))"
+		# these two tests are forced upstream for luajit only
+		rm "${BUILD_DIR}"/regress/{44-resolvers-gc,51-join-defunct-thread}.lua || die
+	else
+		# Thanks to dev-lua/luaossl for this workaround
+		# This is a workaround for luajit, as it confirms to lua5.1
+		# and the 'GNUmakefile' doesn't understand LuaJITs version.
+		LUA_VERSION="5.1"
+	fi
+
+	if [[ ${LUA_VERSION} != 5.3 ]]; then
+		# this test is forced upstream for lua5-3 only
+		rm "${BUILD_DIR}"/regress/152-thread-integer-passing.lua || die
+	fi
+
+	# install tests for lua_version only
+	sed -e 's:for V in 5.1 5.2 5.3 5.4:for V in '${LUA_VERSION}':' \
+		-i "${BUILD_DIR}"/regress/GNUmakefile || die
+
+	popd
+}
+
+src_prepare() {
+	default
+	rm -r vendor || die
+	sed \
+		-e '/HAVE_API_FN =/d' \
+		-e '/ALL_CFLAGS += -g/d' \
+		-e '/ALL_LDFLAGS += -L/d' \
+		-e 's:-O2::' \
+		-i GNUmakefile || die
+
+	# use header from dev-lua/compat53 instead of vendor
```

Comment:
The issue with seds is that they "fail silently" if they stop replacing anything or if they start replacing the wrong thing. It might be better as a patch, unless it'll be huge.

Reply:
ok. I add two patches : rm vendor + qa flags